core/matrix/batch_dense.cpp

```diff
@@ -155,6 +156,31 @@ void BatchDense<ValueType>::add_scaled_impl(const BatchLinOp* alpha,
 }
 
 
+template <typename ValueType>
+void BatchDense<ValueType>::add_scale(const BatchLinOp* const alpha,
+                                      const BatchLinOp* const a,
+                                      const BatchLinOp* const beta)
+{
+    auto batch_alpha = as<BatchDense<ValueType>>(alpha);
+    auto batch_beta = as<BatchDense<ValueType>>(beta);
+    auto batch_a = as<BatchDense<ValueType>>(a);
+    GKO_ASSERT_BATCH_EQUAL_ROWS(
+        batch_alpha, batch_dim<2>(this->get_num_batch_entries(), dim<2>(1, 1)));
+    for (size_type b = 0; b < batch_alpha->get_num_batch_entries(); ++b) {
+        if (batch_alpha->get_size().at(b)[1] != 1) {
+            // different alpha for each column
+            GKO_ASSERT_BATCH_EQUAL_COLS(this, batch_alpha);
+        }
+    }
+    GKO_ASSERT_BATCH_EQUAL_DIMENSIONS(this, batch_a);
+    GKO_ASSERT_BATCH_EQUAL_DIMENSIONS(batch_alpha, batch_beta);
+    auto exec = this->get_executor();
+
+    exec->run(
+        batch_dense::make_add_scale(batch_alpha, batch_a, batch_beta, this));
```

Comment:
nit: as it is being used only once,
```suggestion
    this->get_executor()->run(
        batch_dense::make_add_scale(batch_alpha, batch_a, batch_beta, this));
```